aipg/assistant.py

```
@@ -1,128 +1,132 @@
-            signal.alarm(0)
-            signal.signal(signal.SIGALRM, previous)
-            
-            
-class BaseAssistant[StateT]:
+class BaseAssistant(Generic[StateT]):
     def __init__(self, config: AppConfig) -> None:
         self.config = config
         self.llm = LLMClient(config)
 
     def handle_exception(self, stage: str, exception: Exception):
         raise Exception(f"{stage}: {exception}")
```

Ruff (0.12.2)

Create your own exception

(TRY002)